# EE282 Project Analysis Proposal
## Introduction
The cellular and molecular changes in human glaucoma are important for halting and reversing vision loss but are incompletely understood. This proposal applies the 10X Genomics platform for single nucleus sequencing of primary human donor optic nerve head tissue, a critical site in glaucomatous damage, derived from three glaucomatous and three healthy age-matched controls. These data have been downloaded and files are available on a local diskIn order to draw conclusions about the underlying biological mechanisms at play, these data will require careful bioinformatic analysis in several steps, involving the software tools FastQC, CellRanger, SoupX, Seurat, and cellProfiler, outlined below.
## Count matrix generation and QC
Raw sequencing data will be verified for data integrity and quality to ensure all included data meets expected standards since sequencing errors can masquerade as biological mutations or result in low mapping rates. **FastQC** will be run to verify sequencing reads are reliable, by checking the ‘per base sequence quality’ targeting Q>30. Next, files will be run through the **CellRanger** pipeline which computes the number of unique molecules mapped to the human genome (GR38), as well as identifies the number of barcoded droplets to generate the numbers of reads, cells captured, and genes per cell in a _count matrix_ as its output. Since nuclei sequencing captures unspliced RNAs, intron sequences will be included. Finally, ambient RNA will be subtracted from the dataset using the R software package **SoupX**, essential for removing the ‘soup’ of nonspecific RNAS that are floating in solution with isolated nuclei. 
## Clustering and cell type annotation
The matrices from SoupX will be directed to to **Seurat**, another R package which can filter out cells with >5% mitochondrial reads, since pure nuclei are not expected to contain mitochondria. Seurat also performs clustering and batch correction to yield  visualizations of the high-dimensional data by uniform manifold approximation projection (UMAP). Clusters will be annotated based on plotting markers for expected cell types. Next, the relative abundance of different cell types can be plotted to examine changes in cell composition of glaucomatous human optic nerve head.
## Differential gene expression, pathway analysis
To determine which genes are differentially expressed in my data, I will utilize the MAST function of Seurat, which treats cell clusters as pseudobulk samples. The output of the differential expression can then be fed to **clusterProfiler**, which assigns likely pathway-level changes in basic cell biology and can be visualized using a lollipop chart. CellChat was mentioned in the previous proposal assignment but will not be pursued for this analysis proposal.

## Conclusion
Generation of a pipeline which performs the above features is expected to be a feasible goal that will implement the bioinformatic skills learned in this course and generate novel insight in glaucomatous pathology in human patients. The hypotheses generated from these data may result in the advancement of treatments to prevent and restore blindness.
